File: src/structure/generateImports.ts
import {Configuration} from "../configuration/configuration.js";

export function generateImports(
    outputFileName: string,
    configuration: Configuration,
) {
    const {importInjector} = configuration

    let importSources: string[] = []

    for (const [pattern, imports] of Object.entries(importInjector)) {
        const regexp = new RegExp(pattern)

        if (regexp.test(outputFileName)) {
            importSources = importSources.concat(imports)
        }
    }

    return importSources
        .map(it => `import ${it}`)
        .join("\n")
}

File: gradle-plugin/karakum/src/jsMain/kotlin/io/github/sgrishchenko/karakum/structure/generateImports.kt
package io.github.sgrishchenko.karakum.structure

import io.github.sgrishchenko.karakum.configuration.Configuration
import js.objects.Object

fun generateImports(
    outputFileName: String,
    configuration: Configuration,
): String {
    val importInjector = configuration.importInjector

    val importSources = mutableListOf<String>()

    for ((pattern, imports) in Object.entries(importInjector)) {
        val regexp = pattern.toRegex()

        if (regexp.containsMatchIn(outputFileName)) {
            importSources += imports
        }
    }

    return importSources.joinToString(separator = "\n") { "import $it" }
}
